internal/payloads/error_payload.go

```diff
@@ -15,44 +15,36 @@ type ResponseError struct {
 	// HTTP status code
 	HTTPStatusCode int `json:"-"`
 
-	// error message including message from wrapped error
-	Message string `json:"msg"`
+	// error type name
+	Name string `json:"name"`
```

Comment:
I am wondering if name is a good term here. 
Maybe type is better? what is the purpose of the field?

Reply:
In the future, each error will get its value passed from the place where the error was caught on services. We will have the error itself in the `.Error`, and the `.Message` is supposed to be shown to the user, which will include more details. So I guess the `'Name'` could be handy when we need to use a simple string.
For example, dao error messages tend to be a bit messy. I am sure that Lukas will be able to correct me and clarify this, as we were discussing this before I started working on this. I am unsure if we should use different naming for this. I will probably leave this decision to @lzap.

Reply:
Oh no don't leave those decisions on me, we just brainstormed "name" the other day, I did not put much thought about it.

So the background around this is that we wanted to have all REST error payloads to have:

* Something we can show to the user, a meaningful message like "pubkey upload failed". This can be translated in the UI into other languages via i18n so it should be a short and simple English statement (perhaps not a full sentence).
* Full error as Go returns it (the whole error message chain like `pubkey upload failed: AWS client error: 401 fingerprint already exists`). This would be shown when user clicks on some kind of "Detail".

During a discussion, we were talking about "error codes" and how useful there were in Foreman, we briefly thought about implementing it but then we thought that "error type name" should be better. Now that I think about it, I am not so sure about importance of this information to the user and even for us. Because we also discussed that all our error handling in the application should be done via string errors and wrapping, the de-facto Go standard. At some places we need to check for the type via `errors.As`, but result of this is typically an error value.

So all and all, I think we might actually drop this field completely to be honest, I don't see much value in it.